corehq/apps/locations/views.py

```diff
@@ -407,12 +412,40 @@ def products_form(self):
 
         form = MultipleSelectionForm(
             initial={'selected_ids': self.products_at_location},
-            submit_label=_("Update Product List")
+            submit_label=_("Update Product List"),
+            prefix="products",
         )
         form.fields['selected_ids'].choices = self.all_products
         return form
 
     @property
+    @memoized
+    def all_users(self):
```

Comment:
I looked in to it a bit, and it looks like we do need to load them all each time (there's default validation that checks to make sure the selections are in the list of choices, although we could probably override that).

I did a quick performance test (on a domain with ~400 users) comparing
 * lookup all users in domain using couch (wrapped), filter based on location id, filter for existence of location id
 * lookup all users in elasticsearch, restricting fields returned, lookup users with a specific location in elastic, lookup users with any location assigned

The couch version took `0.52` for one iteration, and `5.2` for 10 iterations.  The elastic version was about an order of magnitude faster, at `0.056` for one iteration, and `0.57` for 10.  I imagine the memory footprint of the elastic one would be considerably better too, but I'm not sure how to test that well.

Still if we really don't trust the results of elastic, then the point is moot anyways.  I do think the user index is a bit more trustworthy than the others, as there's a [signal](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/signals.py#L36-43) which updates individual users in elastic on save (rather than waiting on the pillow).  Not all changes call the `save()` method though.